src/lib.rs

```diff
@@ -101,6 +104,16 @@ impl Client {
     /// ```
     pub fn with_url(mut self, url: impl Into<String>) -> Self {
         self.url = url.into();
+
+        #[cfg(feature = "tls")]
+        if self.url.starts_with("https:") {
```

Comment:
From hyper-tls docs:
> By default this connector will use plain HTTP if the URL provided uses the HTTP scheme (eg: http://example.com/).

Thus, it seems possible to just use `HttpsConnector` instead `HttpConnector` inside `impl Default for Client` if the feature is enabled.